core/src/main/java/com/backblaze/b2/client/B2StorageClientWebifierImpl.java

```diff
@@ -466,9 +468,16 @@ public B2FileVersion getFileInfoByName(B2AccountAuthorization accountAuth,
                 request.getFileName()), makeHeaders(accountAuth));
 
 
-        return new B2FileVersion(headers.getValueOrNull(FILE_ID), headers.getValueOrNull(FILE_NAME),
-                headers.getContentLength(), headers.getContentType(), headers.getContentSha1OrNull(),
-                headers.getB2FileInfo(), "upload", Long.parseLong(headers.getValueOrNull(UPLOAD_TIMESTAMP)));
+        return new B2FileVersion(
+                headers.getValueOrNull(FILE_ID),
+                percentDecode(headers.getValueOrNull(FILE_NAME)),
```

Comment:
what would you think about adding a B2Headers.getFileName() that would do this decode.  it could help prevent other people from forgetting to do the decode.

Reply:
i discussed this with BrianB offline and I'll be making that change.